pkg/operator/secretannotator/openstack/reconciler_test.go

```diff
@@ -220,85 +224,65 @@ func TestReconcileCloudCredSecret_Reconcile(t *testing.T) {
-				wantErr:        false,
+				name:               "No CA Cert",
+				cloudsYAML:         cloudsNoCACert,
+				expectedCloudsYAML: cloudsNoCACert,
 			},
 			{
-				name:           "Incorrect CA Cert",
-				cacert:         incorrectCACertFile,
-				expectedCACert: correctCACertFile,
-				wantErr:        false,
+				name:               "Incorrect CA Cert",
+				cloudsYAML:         fmt.Sprintf(cloudsWithCACert, incorrectCACertFile),
+				expectedCloudsYAML: fmt.Sprintf(cloudsWithCACert, correctCACertFile),
 			},
 			{
-				name:           "Correct CA Cert",
-				cacert:         correctCACertFile,
-				expectedCACert: correctCACertFile,
-				wantErr:        false,
+				name:               "Correct CA Cert",
+				cloudsYAML:         fmt.Sprintf(cloudsWithCACert, correctCACertFile),
+				expectedCloudsYAML: fmt.Sprintf(cloudsWithCACert, correctCACertFile),
 			},
 		} {
 			t.Run(tc.name, func(t *testing.T) {
-				secret := testSecret(tc.cacert)
+				secret := testSecret(tc.cloudsYAML)
 				fakeClient := fake.NewFakeClient(infra, passthrough, secret)
 
+				t.Logf("clouds.yaml: %s", tc.cloudsYAML)
 				r := &ReconcileCloudCredSecret{
 					Client: fakeClient,
 					Logger: log.WithField("controller", "testController"),
 				}
-				_, err := r.Reconcile(reconcile.Request{NamespacedName: types.NamespacedName{
+				_, reconcileErr := r.Reconcile(reconcile.Request{NamespacedName: types.NamespacedName{
 					Name:      "openstack-credentials",
 					Namespace: "kube-system",
 				}})
 
-				if tc.wantErr {
-					require.Error(t, err, "ReconcileCloudCredSecret.Reconcile() did not return expected error")
-				} else {
-					require.NoError(t, err, "ReconcileCloudCredSecret.Reconcile() returned unexpected error")
-				}
-
 				reconciledSecret := &corev1.Secret{}
-				err = fakeClient.Get(context.TODO(), client.ObjectKey{
+				err := fakeClient.Get(context.TODO(), client.ObjectKey{
 					Namespace: secret.Namespace,
 					Name:      secret.Name,
 				}, reconciledSecret)
 				require.NoError(t, err, "Failed to fetch secret after ReconcileCloudCredSecret.Reconcile()")
+				reconciledCloudsYAML := string(reconciledSecret.Data["clouds.yaml"])
 
-				if !tc.skipDiff {
-					// Update cacert in the input secret to the expected value,
-					// and compare the resulting parsed clouds.yamls
-					origClouds, err := parseClouds(secret)
-					require.NoError(t, err, "Unexpected error parsing original clouds.yaml")
-					origClouds.Clouds.OpenStack.CACert = tc.expectedCACert
+				if tc.wantErr {
+					require.Error(t, reconcileErr, "ReconcileCloudCredSecret.Reconcile() did not return expected error")
 
-					reconciledClouds, err := parseClouds(reconciledSecret)
-					require.NoError(t, err, "Unexpected error parsing updated clouds.yaml")
-					assert.Equal(t, origClouds, reconciledClouds, "Secret was not updated as expected")
+					// clouds.yaml should have been modified if there was an error
+					assert.Equal(t, tc.cloudsYAML, reconciledCloudsYAML)
```

Comment:
What is this assertion asserting precisely (maybe put some text in the final parameter to the Equal() call explaining what was expected)?

Reply:
Eurgh! I noted that I hadn't done that yet when I wrote it, and then forgot 🤦‍♂️